packages/view/src/components/Statistics/AuthorBarChart/AuthorBarChart.tsx

```diff
@@ -0,0 +1,123 @@
+import type { ChangeEvent } from "react";
+import { useRef, useEffect, useState } from "react";
+import * as d3 from "d3";
+
+import type { ClusterNode } from "types";
+
+import type { AuthorDataType, MetricType } from "./AuthorBarChart.type";
+import { getDataByAuthor, sortDataByName } from "./AuthorBarChart.util";
+import { DIMENSIONS, METRIC_TYPE } from "./AuthorBarChart.const";
+
+import "./AuthorBarChart.scss";
+
+type AuthorBarChartProps = {
+  data: ClusterNode[];
+};
+
+const AuthorBarChart = ({ data: rawData }: AuthorBarChartProps) => {
+  const svgRef = useRef(null);
+  const [metric, setMetric] = useState<MetricType>(METRIC_TYPE[0]);
+
+  const authorData = getDataByAuthor(rawData);
+  const totalMetricValues = authorData.reduce(
+    (acc, item) => acc + item[metric],
+    0
+  );
+
+  useEffect(() => {
+    const data = authorData.sort((a, b) => {
+      if (a[metric] === b[metric]) {
+        return sortDataByName(a.name, b.name);
+      }
+      return a[metric] - b[metric];
+    });
+
+    const svg = d3
+      .select(svgRef.current)
+      .attr("width", DIMENSIONS.width)
+      .attr("height", DIMENSIONS.height);
+
+    svg.selectAll("*").remove();
+
+    const xAxisGroup = svg
+      .append("g")
+      .attr("class", "axis xAxis")
+      .style("transform", `translateY(${DIMENSIONS.height}px)`);
+    const yAxisGroup = svg.append("g").attr("class", "axis yAxis");
+    const barGroup = svg.append("g").attr("class", "bars");
+
+    // Scales
+    const xScale = d3.scaleLinear().domain([0, 1]).range([0, DIMENSIONS.width]);
+
+    const yScale = d3
+      .scaleBand()
+      .domain(data.map((d) => d.name))
```

Comment:
```suggestion
      .domain(data.map(({ name }) => name))
```
요렇게 디스트럭쳐링 해서 쓰는 거 어떨까요?

Reply:
이 부분은 대부분의 d3 코드에서 그대로 d값을 사용하고 있기도 하고 디스트럭처링 한 변수가 추후 자주 재사용되는 것이 아니라 그대로 진행하고 싶은데 괜찮을까요?

Reply:
(궁금) 진짜 그냥 궁금해서 남기는 질문인데요.

코드를 읽는 그 자체에만 초점을 맞춰보면
`data.map((d) => d.name)` vs `data.map(({ name }) => name)`
↓↓↓
`d를 받아서 "name 만" 가져오겠다` vs `d를 받아오는 중에 "name 만 뽑아서", 그 name 을 가져오겠다` 
이렇게 읽히는 느낌인데요 (개인생각)

특정 속성 하나만 뽑아오는 목적이라면, 저는 개인적으로 전자를 좋아하는데요. 그 이유가 글자수가 더 적어서에요..ㅎㅎ
map 안에 추가적인 매핑로직이 들어간다면 디스트럭쳐링을 사용하는 편이구요.

혹시 다른분들은 어떤 생각이신가요? ㅎㅎ

Reply:
저도 @anpaul0615 님과 비슷한 방식으로 사용하고 있습니다~!
사실 속성 하나만 뽑아 올때 큰 고민없이 전자로 사용했는데 글자수가 적다는 장점이 있다는 건 생각못했네요😂

Reply:
> 코드를 읽는 그 자체에만 초점을 맞춰보면 `data.map((d) => d.name)` vs `data.map(({ name }) => name)` ↓↓↓ `d를 받아서 "name 만" 가져오겠다` vs `d를 받아오는 중에 "name 만 뽑아서", 그 name 을 가져오겠다` 이렇게 읽히는 느낌인데요 (개인생각)

아, 말씀 듣고 보니, `name만 뽑아서`의 장점이 하나 더 있네요!
"다른 property들을 명시적으로 건드리지 않겠다!!" 고 선언해버리는 거군요. 이런건 좋은 것 같습니다.

헌데, 짧아지지만 괄호가 많아져서 읽기 힘들 수도 있겠다는 생각도 또 드네요 ㅎㅎㅎ

Reply:
> 저 궁금한 게 있는데 글자수가 적으면 타이핑하는 횟수가 적기 때문에 개발할 때 효율적이어서 선호하시는 걸까요?

js코드가 적어지면 번들파일 크기도 작아져서 라고 생각하는데... 너무 미세하려나요 ㅋㅋㅋ